mmeval/metrics/coco_detection.py

```diff
@@ -750,6 +750,255 @@ def classes(self) -> list:
+        eval_results['occluded_recall'] = occluded_recall
+        table_results['occluded_recall'] = \
+            f'{round(occluded_recall * 100, 2):.2f}%'
+        table_results['occluded_correct_num'] = f'{occluded_correct_num}'
+
+        self.logger.info('Computing separated mask recall...')
+        separated_correct_num, separated_recall = self.compute_recall(
+            dict_det, gt_ann=self.separated_ann, is_occ=False)
+        self.logger.info(
+            f'COCO separated mask success num: {separated_correct_num}')
+        self.logger.info('COCO separated mask recall: '
+                         f'{round(separated_recall * 100, 2):.2f}%')
+        eval_results['separated_recall'] = separated_recall
+        table_results['separated_recall'] = \
+            f'{round(separated_recall * 100, 2):.2f}%'
+        table_results['separated_correct_num'] = f'{separated_correct_num}'
+
+        if self.print_results:
+            self._print_occluded_separated_recall(table_results)
+
+        return eval_results
+
+    def compute_recall(self,
+                       result_dict: dict,
+                       gt_ann: list,
+                       is_occ: bool = True) -> tuple:
+        """Compute the recall of occluded or separated masks.
+
+        Args:
+            result_dict (dict): Processed mask results.
+            gt_ann (list): Occluded or separated coco annotations.
+            is_occ (bool): Whether the annotation is occluded mask.
+                Defaults to True.
+        Returns:
+            tuple: number of correct masks and the recall.
+        """
+        correct = 0
+        total_ann = len(gt_ann)
+        for iter_i in range(total_ann):
+            cur_item = gt_ann[iter_i]
+            cur_img_name = cur_item[0]
+            cur_gt_bbox = cur_item[3]
+            if is_occ:
+                cur_gt_bbox = [
+                    cur_gt_bbox[0], cur_gt_bbox[1],
+                    cur_gt_bbox[0] + cur_gt_bbox[2],
+                    cur_gt_bbox[1] + cur_gt_bbox[3]
+                ]
+            cur_gt_class = cur_item[1]
+            cur_gt_mask = mask_util.decode(cur_item[4])
+
+            assert cur_img_name in result_dict.keys()
+            cur_detections = result_dict[cur_img_name]
+
+            correct_flag = False
+            for i in range(len(cur_detections)):
+                cur_det_confidence = cur_detections[i][0]
+                if cur_det_confidence < self.score_thr:
+                    break
+                cur_det_class = cur_detections[i][1]
+                if cur_det_class != cur_gt_class:
+                    continue
+                cur_det_mask = cur_detections[i][2]
+                cur_iou = self.mask_iou(cur_det_mask, cur_gt_mask)
+                if cur_iou >= self.iou_thr:
+                    correct_flag = True
+                    break
+            if correct_flag:
+                correct += 1
+            print(
```

Comment:
use a progress bar

Reply:
We can only use rish.progress bar here. Can't use MMEngine.ProgressBar 🤦‍. And I have tried to use rish's, the code will a little bit ugly